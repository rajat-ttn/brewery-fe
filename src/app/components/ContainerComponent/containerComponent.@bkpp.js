// import React from 'react';
// // import { shallow, mount } from 'enzyme';
// import renderer from 'react-test-renderer';
// // import BeerComponent from './index';
// import ContainerComponent from './index';
//
//
// describe('BeerComponent component renders the BeerComponent correctly', () => {
//
//     it('renders correctly with no data', () => {
//         const key = '1';
//         const beerList = [];
//         const rendered = renderer.create(
//             <ContainerComponent beerList={beerList}/>
//         );
//         expect(rendered.toJSON()).toMatchSnapshot();
//     });
// });
