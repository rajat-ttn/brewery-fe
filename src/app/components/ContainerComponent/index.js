import React from 'react';
import BeerComponent from '../BeerComponent'
import './style.css';


const ContainerComponent = ({ beerList }) => {
    return (
        <div className="row">
            {
                beerList && beerList.length ?
                    beerList.map((beer, index) => (
                        <BeerComponent beerContentDetail={beer} key={beer.id}/>
                        )) : null
            }
        </div>
    );
}
export default ContainerComponent;