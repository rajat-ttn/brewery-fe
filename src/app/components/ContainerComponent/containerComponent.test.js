import React from 'react';
// import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
// import BeerComponent from './index';
import ContainerComponent from './index';


describe('BeerComponent component renders the BeerComponent correctly', () => {

    it('renders correctly with no data', () => {
        const key = '1';
        const beerList = [];
        const rendered = renderer.create(
            <ContainerComponent beerList={beerList}/>
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with data', () => {
        const key = '2';
        const beerList = [{ beerType: "Aler Beer", containerId: 1, id: 1, tempRange: [3, 5]}, { beerType: "Aler Beer2", containerId: 2, id: 2, tempRange: [3, 5]},{ beerType: "Aler Beer", containerId: 3, id: 3, tempRange: [3, 5]}];
        const rendered = renderer.create(
            <ContainerComponent beerList={beerList}/>
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });
});
