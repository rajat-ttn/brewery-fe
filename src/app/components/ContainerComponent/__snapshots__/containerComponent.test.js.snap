// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerComponent component renders the BeerComponent correctly renders correctly with data 1`] = `
<div
  className="row"
>
  <div
    className="col-md-4 commonBoxModel"
  >
    <div
      className="beerContainer"
    >
      <figure>
        <img
          alt="Beer 1"
          className="beerContainerImage"
          src="../images/beer.png"
        />
        <figcaption
          className="yellowContainer"
        >
          Aler Beer
          <br />
          Current Temp
          <br />
          Temp Range
        </figcaption>
      </figure>
    </div>
  </div>
  <div
    className="col-md-4 commonBoxModel"
  >
    <div
      className="beerContainer"
    >
      <figure>
        <img
          alt="Beer 1"
          className="beerContainerImage"
          src="../images/beer.png"
        />
        <figcaption
          className="yellowContainer"
        >
          Aler Beer2
          <br />
          Current Temp
          <br />
          Temp Range
        </figcaption>
      </figure>
    </div>
  </div>
  <div
    className="col-md-4 commonBoxModel"
  >
    <div
      className="beerContainer"
    >
      <figure>
        <img
          alt="Beer 1"
          className="beerContainerImage"
          src="../images/beer.png"
        />
        <figcaption
          className="yellowContainer"
        >
          Aler Beer
          <br />
          Current Temp
          <br />
          Temp Range
        </figcaption>
      </figure>
    </div>
  </div>
</div>
`;

exports[`BeerComponent component renders the BeerComponent correctly renders correctly with no data 1`] = `
<div
  className="row"
/>
`;
